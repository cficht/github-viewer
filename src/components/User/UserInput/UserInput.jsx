import React from 'react';
import PropTypes from 'prop-types';

const UserInput = ({ username, onUserChange }) => (
  <>
    <input type="text" name="username" value={username} onChange={onUserChange} />
  </>
);

UserInput.propTypes = {
  username: PropTypes.string.isRequired,
  onUserChange: PropTypes.func.isRequired
};

export default UserInput;
