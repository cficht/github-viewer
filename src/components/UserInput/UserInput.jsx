import React from 'react';
import PropTypes from 'prop-types';

const UserInput = ({ username, onUserChange, onUserSubmit }) => (
  <article>
    <h1>GitHub Viewer</h1>
    <input type="text" name="username" value={username} onChange={onUserChange} />
    <input type="button" name="search" value="Search" onClick={onUserSubmit} />
  </article>
);

UserInput.propTypes = {
  username: PropTypes.string.isRequired,
  onUserChange: PropTypes.func.isRequired,
  onUserSubmit: PropTypes.func.isRequired
};

export default UserInput;
