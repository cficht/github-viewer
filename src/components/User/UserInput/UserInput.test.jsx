import React from 'react';
import { shallow } from 'enzyme';
import UserInput from './UserInput';

describe('UserInput component', () => {
  it('renders UserInput', () => {
    const wrapper = shallow(<UserInput 
      username="cficht"
      onUserChange={() => {}}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
