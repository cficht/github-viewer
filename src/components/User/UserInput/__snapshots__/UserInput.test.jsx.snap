// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInput component renders UserInput 1`] = `
<Fragment>
  <input
    name="username"
    onChange={[Function]}
    type="text"
    value="cficht"
  />
</Fragment>
`;
